Resources/Storage/GFAL2_StorageBase.py

```diff
@@ -710,6 +713,9 @@ def __getSingleMetadata( self, path ):
           metadataDict['Unavailable'] = 1
         else:
           metadataDict['Unavailable'] = 0
+        if attributeDict['user.status'] == 'ONLINE_AND_NEARLINE':
```

Comment:
First of all this whole part can be very much factorized. Something like
user_status = attributeDict.get('user.status', '')
metadataDict['Cached'] = int('ONLINE' in user_status)
metadataDict['Migrated'] = int('NEARLINE' in user_status)
metadataDict['Lost'] = int(user_status == 'LOST')
metadataDict['Unavailable'] = int(user_status == 'UNAVAILABLE')

Second, all these status are returned by SRM. Does that also work for the other protocol ? The 4 entries in the dict must be present, but if these status (nearline, and so on) are srm specific, then they should be in the srm plugin, not the base class)